src/lib.rs

```diff
@@ -396,4 +432,242 @@ mod test {
+
+        assert_tokens(
+            &Verbosity::<WarnLevel>::new(0, 2),
+            &[Token::UnitVariant {
+                name: "LevelFilter",
+                variant: "OFF",
+            }],
+        );
+
+        assert_tokens(
+            &Verbosity::<WarnLevel>::new(0, 1),
+            &[Token::UnitVariant {
+                name: "LevelFilter",
+                variant: "ERROR",
+            }],
+        );
+
+        assert_tokens(
+            &Verbosity::<WarnLevel>::new(0, 0),
+            &[Token::UnitVariant {
+                name: "LevelFilter",
+                variant: "WARN",
+            }],
+        );
+
+        assert_tokens(
+            &Verbosity::<WarnLevel>::new(1, 0),
+            &[Token::UnitVariant {
+                name: "LevelFilter",
+                variant: "INFO",
+            }],
+        );
+
+        assert_tokens(
+            &Verbosity::<WarnLevel>::new(2, 0),
+            &[Token::UnitVariant {
+                name: "LevelFilter",
+                variant: "DEBUG",
+            }],
+        );
+
+        assert_tokens(
+            &Verbosity::<WarnLevel>::new(3, 0),
+            &[Token::UnitVariant {
+                name: "LevelFilter",
+                variant: "TRACE",
+            }],
+        );
+
+        assert_tokens(
+            &Verbosity::<InfoLevel>::new(0, 3),
+            &[Token::UnitVariant {
+                name: "LevelFilter",
+                variant: "OFF",
+            }],
+        );
+
+        assert_tokens(
+            &Verbosity::<InfoLevel>::new(0, 2),
+            &[Token::UnitVariant {
+                name: "LevelFilter",
+                variant: "ERROR",
+            }],
+        );
+
+        assert_tokens(
+            &Verbosity::<InfoLevel>::new(0, 1),
+            &[Token::UnitVariant {
+                name: "LevelFilter",
+                variant: "WARN",
+            }],
+        );
+
+        assert_tokens(
+            &Verbosity::<InfoLevel>::new(0, 0),
+            &[Token::UnitVariant {
+                name: "LevelFilter",
+                variant: "INFO",
+            }],
+        );
+
+        assert_tokens(
+            &Verbosity::<InfoLevel>::new(1, 0),
+            &[Token::UnitVariant {
+                name: "LevelFilter",
+                variant: "DEBUG",
+            }],
+        );
+
+        assert_tokens(
+            &Verbosity::<InfoLevel>::new(2, 0),
+            &[Token::UnitVariant {
+                name: "LevelFilter",
+                variant: "TRACE",
+            }],
+        );
+    }
+
+    #[test]
+    #[cfg(feature = "serde")]
+    fn serde_toml() {
+        use clap::Parser;
+        use serde::{Deserialize, Serialize};
+
+        #[derive(Debug, Parser, Serialize, Deserialize)]
+        struct Cli {
+            meaning_of_life: u8,
+            #[command(flatten)]
+            verbose: Verbosity<InfoLevel>,
+        }
+
+        // round-trips
+        let toml = "meaning_of_life = 42\nverbose = \"DEBUG\"\n";
```

Comment:
Not sure how I feel about deserializing to all-caps

Reply:
Yeah, this irks me a little too, but being consistent with the existing behavior won out over my ideal approach. If I was designing the `log` crate, I think I'd have made that serialization lower case too.

Here, this can deserialize from either case, but serializes to upper case. It comes from the serialization of `log::LevelFilter`. That means upper case is consistent with any other serialization that works with log level serialization (and is consistent with the usual display of the level in log outputs).

If you did want to avoid this, it would require going back to using a custom serialization implementation. This doesn't seem worth the extra code and tests to me. I prefer the simplicity even if that comes with a slightly ugly default which is consistent with other things doing the same thing.

I also suspect the serialization side of this will be used much less frequently than the deserialization side, and the serialization side will also often be more focused on programs re-reading the format rather than humans.

Reply:
On one hand, *if* people are serializing `LevelFilter`, this keeps us consistent with that.  On the other hand, this will produce results inconsistent with every other serialized value.  This isn't just important here but for other cases like schemars.  I think I want to let this sit for a time to mull over this.

Reply:
no problem :)

Reply:
eb58de2d699758a72b437a83059d01c9179aa407 serializes to `VerbosityFilter` instead of this, which doesn't have a deserialization implementation that handles upper case tokens.

Reply:
That replaced SCREAMING CASE with Title case
```rust
assert_eq!(toml, "meaning_of_life = 42\nverbosity = \"Debug\"\n");
```
I feel like lower case is much more likely to align with a users configuration format design (as most are likely kebab-case or snake_case)

Reply:
btw eb58de2  just changed the value but it doesn't answer the question about interoperability